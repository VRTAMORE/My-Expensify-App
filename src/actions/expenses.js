import { v4 as uuidv4 } from 'uuid';
import { ref, remove, set, update, get, onChildChanged, off, push } from 'firebase/database';

import db from '../firebase/firebase';

//ADD_EXPENSE
export const addExpense = (expense) => ({
    type: 'ADD_EXPENSE',
    expense
});

export const startAddExpense = (expenseData = {}) => {
    return (dispatch) => {
        const {
            description = '',
            note = '',
            amount = 0,
            createdAt = 0
        } = expenseData;

        const expense = { description, note, amount, createdAt };
        return push(ref(db, 'expenses'),
            expense
        ).then((ref) => {
            dispatch(addExpense({
                id: ref.key,
                ...expense
            }));
        });
    };
};

//REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
});

//EDIT_EXPENSE
export const editExpense = (id, update) => ({
    type: 'EDIT_EXPENSE',
    id,
    update
});

