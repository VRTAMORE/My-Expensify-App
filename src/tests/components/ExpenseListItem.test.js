import React from 'react';
import { shallow } from 'enzyme';

import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('Render ExpenseListItem with fixture data', () => {
    const exp = expenses[2];
    const wrapper = shallow(<ExpenseListItem
        {...exp}
    />);
    expect(wrapper).toMatchSnapshot();
});