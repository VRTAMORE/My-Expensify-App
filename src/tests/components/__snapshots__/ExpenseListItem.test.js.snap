// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ExpenseListItem with fixture data 1`] = `
<div>
  <Link
    to="/edit/3"
  >
    <h3>
      Credit Card
    </h3>
  </Link>
  <p>
    $45.00
    -
    1st January, 1970
  </p>
</div>
`;
